### ISSUE TITLE
[Bugs] PHP Staticpage rendering different in 1.5.0b1

### ISSUE THREAD

**Issue by mystralkk** [Geeklog-Core/geeklog#119]
**Reported by mst3kroqs on 10 May 2008 22:39**

[Mantis Bugtracker #628](http://project.geeklog.net/tracking/view.php?id=628)

**Comment by mystralkk** [comment]
**Comment posted by dhaun on 11 May 2008 07:59**

I assume the intention with the printer and edit icons was to put them in roughly the same place as they are in a story. Since the block header is different (and can be switched off for static pages) that didn&#039;t work out.

I&#039;d put them in the footer then, next to the change date and hits counter, but couldn&#039;t make them to line up properly.

Suggestions?

**Comment by mystralkk** [comment]
**Comment posted by mst3kroqs on 11 May 2008 13:00**

Hi Dirk - First, it gets more interesting.  When I hide the date, hits and the printer icon, the &#039;info_separator&#039; (vertical bar) is still shown.  This is also an artifact of how the rendering is done now.

The fix for the icons is basically just a change to the sp template file.  The fix for the above is a change to the rendering logic -  I&#039;m working on a patch, will submit for your perusal later today.

**Comment by mystralkk** [comment]
**Comment posted by mst3kroqs on 11 May 2008 13:53**

Further note to the above - the info_separator is only shown on FireFox.  Apparently, FireFox is ignoring the style visibility: hidden; directive.  (sigh)

Update:  This was caused by a bug in the original template - two right braces, eg. 

{info_separator}}

**Comment by mystralkk** [comment]
**Comment posted by dhaun on 11 May 2008 14:03**

The extra } was already fixed in CVS yesterday :)

**Comment by mystralkk** [comment]
**Comment posted by mst3kroqs on 11 May 2008 16:18**

Uploading proposed fix to:

/plugins/staticpages/templates/staticpage.thtml
/plugins/staticpages/functions.inc

Changes:

Move {print_icon}, {edit_icon} after {lastupdate} {hits}, and allow separator to be defined.  Added a couple of vars to control visibility of separators should anyone want to do that, and added logic to SP_displayPage to toggle those as needed.  The separators are null in the updated template.

BTW - $icons always evaluates to &quot;&quot; in SP_displayPage, not sure what is going on with:  $spage-&gt;set_var(&#039;icons&#039;, $icons);

Lastly - $_SP_CONF[&#039;in_block&#039;] seems to be used in /plugins/staticpages/services.inc.php, but not in any of the rendering logic.  Is this intentional?

Hope this helped!

**Comment by mystralkk** [comment]
**Comment posted by dhaun on 12 May 2008 09:24**

I&#039;ve moved the print and edit icons to the bottom of the page now, but didn&#039;t use your modifications. We don&#039;t have separators between icons elsewhere either, so this would only introduce inconsistencies in the layout.

My earlier problems of placing them were caused by the class=&quot;story-icons&quot; - simply leaving this out positions them nicely.

Have to look into the other issues you mentioned ...

(edited to fix typos - need more coffee)

**Comment by mystralkk** [comment]
**Comment posted by dhaun on 12 May 2008 10:31**

Removed the {icons} variable. There was no such variable in 1.4.1, it&#039;s not used in 1.5.0 and it&#039;s always empty anyway. We have separate variables for the print and edit icons, so we don&#039;t need it.

**Comment by mystralkk** [comment]
**Comment posted by dhaun on 12 May 2008 10:56**

$_SP_CONF[&#039;in_block&#039;] is the default setting for the &quot;wrap page in block&quot; option, so it&#039;s only used in the editor and when a static page is created through the webservices interface. So this is fine.

I noticed that the template for the centerblocks had the same issues with the icons, plus it used {lastupdate} and {hits} in the template file but they weren&#039;t defined for centerblocks.

In the end, I made sure those variables are defined there, but decided that you wouldn&#039;t normally want to have them (or the printer icon) on a centerblock anyway, so I stripped down the centerblock template file to just the {content}.
